### ISSUE TITLE
GCLK.H dependency

### ISSUE THREAD

**Issue by ebenl** [ee-quipment/ZeroPowerManager#2]
When compiling from Arduino IDE, I get this compile error :

....ZeroPowerManager.c:49:19: fatal error: gclk.h: No such file or directory

This is using the Adafruit Feather M0 board and Arduino 1.8.8

**Comment by fabienroots** [comment]
I have the same issue, have found a solution ?

**Comment by jpmeijers** [comment]
Same problem here. If I comment out that line I get another error, so Arduino also does not include that header elsewhere.

`ZeroPowerManager.c:125:35: error: storage size of 'config' isn't known
     struct system_gclk_gen_config config;`

**Comment by ebenl** [comment]
> Same problem here. If I comment out that line I get another error, so Arduino also does not include that header elsewhere.
> 
> `ZeroPowerManager.c:125:35: error: storage size of 'config' isn't known struct system_gclk_gen_config config;`

I ended up not using this library.

Basically :


// Disable USB
 USB->DEVICE.CTRLA.reg &= ~USB_CTRLA_ENABLE;
   USBDevice.detach();
   
      
        //Serial1.end();
    
    
        
        SYSCTRL->BOD33.reg = 0;  //incase...
        
  
       // PM->SLEEP.bit.IDLE = 0;
        SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;/* Enable deepsleep */
       
    
        __DSB(); // Finish when CPU write buffer completed
        __WFI();/* Enter sleep mode */


The sleep current varies depending on the temperature. I measure between 5uA and 10uA

**Comment by jpmeijers** [comment]
Putting the microcontroller into sleep is easy. And there are many libraries to use for it. What I want to do is to drop the main clock frequency from 48MHz down to one of the lower ones. This is the only library I found that can do it. I'm trying to figure out how to do it manually, but the clock chains are very complicated.

**Comment by ee-quipment** [comment]
Modified ZeroPowerManager.c to remove dependency on gclk.h. Includes sam.h now, but the Arduino IDE should be able to find that file. Tested on a fresh installation of Arduino IDE 1.8.13 with an Adafruit M0 Express board.
